addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/data/SeverityComparator.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.dwdunwetter.internal.data;
+
+import java.util.Comparator;
+
+import org.apache.commons.lang.ObjectUtils;
+
+/**
+ * Comperator to sort a Warning first by Severity, second by the onSet date.
+ *
+ * @author Martin Koehler - Initial contribution
+ */
+public class SeverityComparator implements Comparator<DwdWarningData> {
+
+    @Override
+    public int compare(DwdWarningData o1, DwdWarningData o2) {
```

Comment:
Can be replaced by a inline field at the usage site:
```
Comparator.comparing(d -> d.getSeverity().getOrder()).thenComparing(DwdWarningDatagetOnset());
```

Reply:
That doesn't really work. I think because getOrder returns an int, I can't chain the functions. I might get it to work, but it would create an ugly code.

Reply:
`.thenComparingInt()` should work.